src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -65,27 +65,48 @@ public static unsafe void Fill<T>(ref T refData, nuint numElements, T value)
                     {
                         vector = Vector256.Create(vec128).AsVector();
                     }
+                    else if (Vector<byte>.Count == 64)
+                    {
+                        vector = Vector512.Create(vec128).AsVector();
+                    }
                     else
                     {
-                        Debug.Fail("Vector<T> isn't 128 or 256 bits in size?");
+                        Debug.Fail("Vector<T> is unexpected size.");
                         goto CannotVectorize;
                     }
                 }
                 else if (sizeof(T) == 32)
                 {
+                    Vector256<byte> vec256 = Unsafe.As<T, Vector256<byte>>(ref tmp);
                     if (Vector<byte>.Count == 32)
                     {
-                        vector = Unsafe.As<T, Vector256<byte>>(ref tmp).AsVector();
+                        vector = vec256.AsVector();
+                    }
+                    else if (Vector<byte>.Count == 64)
+                    {
+                        vector = Vector512.Create(vec256).AsVector();
+                    }
+                    else
+                    {
+                        Debug.Fail("Vector<T> is unexpected size.");
+                        goto CannotVectorize;
+                    }
+                }
+                else if (sizeof(T) == 64)
+                {
+                    if (Vector<byte>.Count == 64)
+                    {
+                        vector = Unsafe.As<T, Vector512<byte>>(ref tmp).AsVector();
```

Comment:
here

Reply:
That's not adding a Vector512<T> path, it's handling Vector<T> being the same size as Vector512<T>. There is a difference.

Reply:
> I mean these changes, that add a `Vector512` path to `SpanHelpers.Fill`.

This is just unblocking the opt-in for 512-bit Vector<T>, which previously either crashed (debug) or fell back to scalar here.

Notably, `Fill` is an intrinsic which has a different optimized implementation JIT-side.  This managed fallback is less important to optimize, but if you feel like making a PR to do that, it can be evaluated separately.